qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -899,7 +899,18 @@ def _inject_userscripts(self):
             # @run-at (and @include/@exclude/@match) is parsed by
             # QWebEngineScript.
             new_script = QWebEngineScript()
-            new_script.setWorldId(QWebEngineScript.MainWorld)
+            try:
+                world = int(script.jsworld)
+            except ValueError:
+                try:
+                    world = [v for k, v in _JS_WORLD_MAP.items()
+                             if k.name == script.jsworld.lower()][0]
+                except IndexError:
+                    log.greasemonkey.error(
+                        "script {} has invalid value for '@qute-js-world'"
+                        ": {}".format(script.name, script.jsworld))
+                    continue
+            new_script.setWorldId(world)
```

Comment:
You should probably do the same in `WebEngineView._inject_userjs` as well for Qt 5.7.1... though thinking about it, this should really be moved to `_WebEngineScripts` too, it can probably reuse some code from here then. I might take a look at it somewhere over the next days unless you're faster.